src/main/java/uk/gov/laa/ccms/caab/service/AssessmentService.java

```diff
@@ -447,137 +485,640 @@ private boolean differenceGreaterThanTenSecs(final Date keyChangeDate, final Dat
   }
 
   /**
-   * Retrieves the most recent date on which any key data was modified within the given
-   * application. This includes checks across proceedings and opponents involved in the
+   * Cleans up data by deleting redundant opponents and proceedings from the assessment.
+   *
+   * @param assessment the assessment detail to clean up
+   * @param application the application detail for reference
+   */
+  public void cleanupData(
```

Comment:
Please can these be compared to the files:

com.ezgov.legalservices.ccms.utils.IframeHelper In PUI

The logic should be comparable to:
public boolean cleanupData(OpaSession opaSession, Case app) {

It has been cleaned up and duplication removed where possible